### PR TITLE
media-video/openshot: bump to 2.4.0 + bump libs + remove myself from maintainers

### PR DESCRIPTION
**Pull request by sbraz**
I don't really use OpenShot and upstream doesn't seem to be very responsive to user feedback (they haven't closed an issue or a PR on GH for 5 months) so I'm going to stop maintaining this.
I'm hoping someone will be willing to take over :)
There wasn't much to do except updating the patch which fixes the path to the tests.

```diff
--- libopenshot-0.1.7.ebuild    2017-09-02 14:45:01.181385729 +0200                                      
+++ libopenshot-0.1.8.ebuild    2017-09-19 21:56:18.903321959 +0200                                      
@@ -38,7 +38,7 @@                                   
 "                                                  
                                                    
 # https://github.com/OpenShot/libopenshot/pull/45  
-PATCHES=( ${FILESDIR}/${PN}-0.1.3-fix-tests.patch )                                                     
+PATCHES=( ${FILESDIR}/${PN}-0.1.8-fix-tests.patch )                                                     
                                                    
 pkg_pretend() {                                    
        if [[ ${MERGE_TYPE} != binary ]] && ! tc-has-openmp; then                                        
```

```diff
--- openshot-2.3.4.ebuild       2017-09-02 14:45:01.291386333 +0200                                      
+++ openshot-2.4.0.ebuild       2017-09-19 21:56:18.903321959 +0200                                      
@@ -22,7 +22,7 @@                                   
                                                    
 RDEPEND="                                          
        dev-python/PyQt5[svg,webkit,${PYTHON_USEDEP}]                                                    
-       >=media-libs/libopenshot-0.1.7[python,${PYTHON_USEDEP}]                                          
+       >=media-libs/libopenshot-0.1.8[python,${PYTHON_USEDEP}]                                          
        dev-python/httplib2[${PYTHON_USEDEP}]       
        dev-python/pyzmq[${PYTHON_USEDEP}]          
 "                                                  
```

### CODE REVIEW

File: media-libs/libopenshot/libopenshot-0.1.8.ebuild [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+PATCHES=( ${FILESDIR}/${PN}-0.1.8-fix-tests.patch )
+
+pkg_pretend() {
+	if [[ ${MERGE_TYPE} != binary ]] && ! tc-has-openmp; then
+		eerror "${P} requires a compiler with OpenMP support. Your current"
+		eerror "compiler does not support it. If you use gcc, you can"
+		eerror "re-emerge it with the 'openmp' use flag enabled."
+		die "The current compiler does not support OpenMP"
+	fi
+}
```
Comment (SoapGentoo) [inline media-libs/libopenshot/libopenshot-0.1.8.ebuild]:
```
pkg_pretend() {
	[[ ${MERGE_TYPE} != binary ]] && tc-check-openmp
}
```

File: media-libs/libopenshot/libopenshot-0.1.8.ebuild [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+pkg_setup() {
+	use python && python-single-r1_pkg_setup
```
Comment (SoapGentoo) [inline media-libs/libopenshot/libopenshot-0.1.8.ebuild]: add below here:
`[[ ${MERGE_TYPE} != binary ]] && tc-check-openmp` (yes you heard right, it needs to be repeated in `pkg_setup` too)

File: media-libs/libopenshot/libopenshot-0.1.8.ebuild [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+src_prepare() {
+	# https://github.com/OpenShot/libopenshot/issues/17
+	use test || cmake_comment_add_subdirectory tests
+	cmake-utils_src_prepare
```
Comment (SoapGentoo) [inline media-libs/libopenshot/libopenshot-0.1.8.ebuild]: call `cmake-utils_src_prepare` as the first thing

File: media-libs/libopenshot/libopenshot-0.1.8.ebuild [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+src_configure() {
+	local mycmakeargs=(
+		-DENABLE_RUBY=OFF # TODO: add ruby support
+		-DENABLE_PYTHON=$(usex python)
+		-DCMAKE_DISABLE_FIND_PACKAGE_ImageMagick=$(usex !imagemagick)
```
Comment (SoapGentoo) [inline media-libs/libopenshot/libopenshot-0.1.8.ebuild]: there's a nicer command for this: `$(cmake-utils_use_find_package imagemagick)` (it gets the negative logic right so you dont have to)

File: media-video/openshot/openshot-2.4.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+	dev-python/setuptools[${PYTHON_USEDEP}]
+"
+
+S="${WORKDIR}"
+
+src_prepare() {
+	default
+	# prevent setup.py from trying to update MIME databases
+	sed -i 's/^ROOT =.*/ROOT = False/' setup.py || die
+}
```
Comment (SoapGentoo) [inline media-video/openshot/openshot-2.4.0.ebuild]: as is usual for python packages: don't override `src_prepare`, but rather `python_prepare_all`

**Comment by monsieurp** [comment]
Merged.

**Pull request closed without merging**